src/subscription.js

```diff
@@ -137,6 +147,7 @@ class Subscription extends EventEmitter {
         // `stop()` was called before the subscription was ready.
         reject(new Error('Subscription is already stopped'));
       } else {
+        this._whenReadyPending = true;
```

Comment:
Why introduce another variable? When should `this._whenReadyResolveFn` and `this._whenReadyRejectFn` have a value and it not be in a 'pending' state?

Reply:
I was also thinking about it, but for some reason we don't set them to `null` after they are triggered once.

Reply:
Oh ok. Would it make sense to do so?

Reply:
I believe both approaches are good to go. We can either use only these function values and set them to `null` in needed places (but not in `reset`, as `whenReady` might be in progress). Or we can use another variable which explicitly says that we still have `whenReady` pending state.

I believe it's a matter of taste here. Only drawback that I see in using function values as an indicator is that we need to reset them both each time (or add another function to reset them):

```js
this._whenReadyResolveFn = null;
this._whenReadyRejectFn = null;
```

But I like this approach, as we can fully remove resetting of these functions from `_reset`. Because if they have value, we don't need to remove them (`whenReady` is pending), and if they don't have value, then it makes no sense to remove it again :)

Will update and rerequest your review.